src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferDictionaryTryGetValueAnalyzer.cs

```diff
@@ -0,0 +1,145 @@
+            if (containsKeyInvocation is not null && containsKeyInvocation.TargetMethod.Name == ContainsKeyMethodName)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static IInvocationOperation? FindContainsKeyInvocation(IOperation baseOperation)
+        {
+            return baseOperation switch
+            {
+                IInvocationOperation i when i.TargetMethod.Name == ContainsKeyMethodName => i,
+                IBinaryOperation { OperatorKind: BinaryOperatorKind.ConditionalAnd or BinaryOperatorKind.ConditionalOr } b =>
+                    FindContainsKeyInvocation(b.LeftOperand) ?? FindContainsKeyInvocation(b.RightOperand),
+                _ => null
+            };
+        }
+
+        private static bool DictionaryEntryIsModified(IPropertyReferenceOperation dictionaryAccess, IBlockOperation blockOperation)
+        {
+            return blockOperation.Operations.OfType<IExpressionStatementOperation>().Any(o =>
+                o.Operation is IAssignmentOperation { Target: IPropertyReferenceOperation reference } && reference.Property.Equals(dictionaryAccess.Property, SymbolEqualityComparer.Default));
+        }
+
+        private static bool IsDictionaryAccess(IPropertyReferenceOperation propertyReference, INamedTypeSymbol dictionaryType)
+        {
+            return propertyReference.Property.IsIndexer && IsDictionaryType(propertyReference.Property.ContainingType, dictionaryType) &&
+                   (propertyReference.Property.OriginalDefinition.Name == IndexerName || propertyReference.Language == LanguageNames.VisualBasic && propertyReference.Property.OriginalDefinition.Name == IndexerNameVb);
+        }
+
+        private static bool TryGetParentConditionalOperation(IOperation derivedOperation, [NotNullWhen(true)] out IConditionalOperation? conditionalOperation)
+        {
+            conditionalOperation = null;
+            do
+            {
+                if (derivedOperation.Parent is IConditionalOperation c)
+                {
+                    conditionalOperation = c;
+
+                    return true;
+                }
+
+                derivedOperation = derivedOperation.Parent;
+            } while (derivedOperation.Parent != null);
+
+            return false;
+        }
+
+        private static bool IsDictionaryType(INamedTypeSymbol derived, ISymbol dictionaryType)
+        {
+            var constructedDictionaryType = derived.GetBaseTypesAndThis()
+                .WhereAsArray(x => x.OriginalDefinition.Equals(dictionaryType, SymbolEqualityComparer.Default))
+                .SingleOrDefault() ?? derived.AllInterfaces
+                .WhereAsArray(x => x.OriginalDefinition.Equals(dictionaryType, SymbolEqualityComparer.Default))
+                .SingleOrDefault();
+
+            return constructedDictionaryType is not null;
+        }
+
+        private static LocalizableString CreateResource(string resourceName)
```

Comment:
Normally we do static import of `MicrosoftNetCoreAnalyzersResources` 
https://github.com/dotnet/roslyn-analyzers/blob/04edea1ac503805a8d9a6959ec18e20bff506bb5/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AttributeStringLiteralsShouldParseCorrectly.cs#L15
and reuse `CreateLocalizableResourceString`
https://github.com/dotnet/roslyn-analyzers/blob/04edea1ac503805a8d9a6959ec18e20bff506bb5/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AttributeStringLiteralsShouldParseCorrectly.cs#L27

Reply:
Done.